torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -393,12 +393,30 @@ class ScriptModuleSerializer {
     // Then we serialize all code info.
     writeCode(module.type());
     // The tensor constants from the code are written to a separate archive
-    // so loading the code does not depend on loading the data
+    // (constant archive) so loading the code does not depend on loading the
+    // data
     std::vector<IValue> ivalue_constants(
         constant_table_.begin(), constant_table_.end());
+    TensorIndexMap constants_from_jit;
+
+    // Use constants_from_jit to store the mapping (tensor) -> (archive_name,
+    // index), such that when later writing bytecode archive, if the tensor is
+    // the same as one from jit, it will just update bytecode.pkl and won't
+    // write the tensor in bytecode folder.
+    for (size_t i = 0; i < ivalue_constants.size(); i++) {
+      if (ivalue_constants[i].isTensor() &&
+          constants_from_jit.find(ivalue_constants[i].toTensor()) ==
+              constants_from_jit.end()) {
+        constants_from_jit[ivalue_constants[i].toTensor()] =
+            std::make_pair("constants", i);
```

Comment:
Can you add `constexpr std::string kConstantsArchiveName = "constants"` and replace string "constants" with that?

Reply:
Update it to `static constexpr const char* kArchiveNameConstants = "constants";` in the latest commit.